assets/global.js

```diff
@@ -543,13 +543,28 @@ class SliderComponent extends HTMLElement {
       this.nextButton.removeAttribute('disabled');
     }
 
+    if (this.currentPage != previousPage) {
+      this.dispatchEvent(new CustomEvent('slideChanged', { detail: {
+        currentPage: this.currentPage
+      }}));
+    }
+
+    if (!this.pageCount || !this.pageTotal) return;
+
     this.pageCount.textContent = this.currentPage;
     this.pageTotal.textContent = this.totalPages;
   }
 
+  scrollIntoView(element) {
+    const lastVisibleSlide = this.slider.clientWidth + this.slider.scrollLeft - 10;
+    const isSlideVisible = element.offsetLeft < lastVisibleSlide && element.offsetLeft > this.slider.scrollLeft;
+    if (!isSlideVisible) this.slider.scrollTo({ left: element.offsetLeft });
+  }
+
   onButtonClick(event) {
     event.preventDefault();
-    const slideScrollPosition = event.currentTarget.name === 'next' ? this.slider.scrollLeft + this.sliderLastItem.clientWidth : this.slider.scrollLeft - this.sliderLastItem.clientWidth;
+    const step = event.currentTarget.dataset.step || 1;
+    const slideScrollPosition = event.currentTarget.name === 'next' ? this.slider.scrollLeft + (step * this.sliderLastItem.clientWidth) : this.slider.scrollLeft - (step * this.sliderLastItem.clientWidth);
```

Comment:
There is a bit of a delay happening sometimes between the thumbnail sliding to the currently active image: [video](https://screenshot.click/21-11-egsr0-gcq7m.mp4)

Reply:
I have no idea how to fix this without a big refactor on the slider component.

I don't think it's blocker, though. Would it be okay to solve this on follow up PRs?